rct229/rulesets/ashrae9012019/ruleset_functions/get_building_segment_swh_bat.py

```diff
@@ -33,63 +36,61 @@ def get_building_segment_swh_bat(
-        building_segment_id,
-        rmd,
+    building_segment = find_exactly_one_building_segment(rmd, building_segment_id)
+
+    building_segment_swh_bat = building_segment.get(
+        "service_water_heating_building_area_type", None
     )
 
-    if building_segment.get("service_water_heating_building_area_type"):
-        building_segment_swh_bat = building_segment[
-            "service_water_heating_building_area_type"
-        ]
-    else:
+    if building_segment_swh_bat is None:
         swh_use_dict = {}
         for swh_use in find_all(
             f'$.buildings[*].building_segments[*][?(@.id="{building_segment["id"]}")].zones[*].spaces[*].service_water_heating_uses[*]',
             rmd,
         ):
-            swh_use = find_exactly_one_with_field_value(
-                f"$.zones[*].spaces[*].service_water_heating_uses[*]",
-                "id",
-                swh_use["id"],
-                building_segment,
-            )
-
-            if swh_use.get("use_units") == SERVICE_WATER_HEATING_USE_UNIT.OTHER:
-                return "UNDETERMINED"
+            if (
+                swh_use.get("use_units", SERVICE_WATER_HEATING_USE_UNIT.OTHER)
+                == SERVICE_WATER_HEATING_USE_UNIT.OTHER
+            ):
+                building_segment_swh_bat = "UNDETERMINED"
+            else:
+                swh_use_energy_by_space = get_energy_required_to_heat_swh_use(
+                    swh_use["id"], rmd, building_segment["id"], is_leap_year
+                )
 
-            swh_use_energy_by_space = get_energy_required_to_heat_swh_use(
-                swh_use["id"], rmd, building_segment["id"], is_leap_year
-            )
+                if swh_use.get("area_type"):
+                    area_type = swh_use["area_type"]
+                    swh_use_dict.setdefault(area_type, ZERO.ENERGY)
+                    swh_use_dict[area_type] += sum(swh_use_energy_by_space.values())
+                else:
+                    for space_id in swh_use_energy_by_space:
+                        if space_id != "no_spaces_assigned":
+                            space = find_exactly_one_space(rmd, space_id)
+                            if space.get("service_water_heating_building_area_type"):
+                                service_water_heating_bat = space[
+                                    "service_water_heating_building_area_type"
+                                ]
+                                swh_use_dict.setdefault(
+                                    service_water_heating_bat, ZERO.ENERGY
+                                )
+                                swh_use_dict[
+                                    service_water_heating_bat
+                                ] += swh_use_energy_by_space[space_id]
+                            else:
+                                swh_use_dict.setdefault("UNDETERMINED", ZERO.ENERGY)
+                                swh_use_dict["UNDETERMINED"] += swh_use_energy_by_space[
```

Comment:
One edge case here is that the value of a key can be None in the return dict of [get_energy_required_to_heat_swh_use](https://github.com/pnnl/ruleset-checking-tool/blob/4093067ad2d734bdbbf2259c639160ddb2ddf04e/rct229/rulesets/ashrae9012019/ruleset_functions/get_energy_required_to_heat_swh_use.py#L40)

Reply:
Addressed.

Reply:
Same here.